blocks/gnav/gnav.js

```diff
@@ -1,287 +1,414 @@
+   */
+  toggleMenu = (el) => {
+    const isSearch = el.classList.contains('gnav-search');
+    const sameMenu = el === this.state.openMenu;
+    if (this.state.openMenu) {
+      this.closeMenu();
+    }
+    if (!sameMenu) {
+      this.openMenu(el, isSearch);
     }
-    sectionEl.appendChild(navItemEl);
-  });
+  }
 
-  return (gnav);
-}
+  closeMenu = () => {
+    this.state.openMenu.classList.remove(IS_OPEN);
+    document.removeEventListener('click', this.closeOnDocClick);
+    window.removeEventListener('keydown', this.closeOnEscape);
+    this.curtain.classList.remove(IS_OPEN);
+    this.state.openMenu = null;
+  }
 
-async function markupToNav(url) {
-  const resp = await fetch(`${url}.plain.html`);
-  const html = await resp.text();
-  const header = document.createElement('header');
-  header.innerHTML = html;
-  const nav = {};
-  nav.top = [...header.querySelectorAll(':scope > div h2')].map((h2) => {
-    const navItem = {};
-    const div = h2.closest('div');
-    navItem.text = h2.textContent;
-    const h2a = h2.closest('a') || h2.querySelector('a');
-    if (h2a) {
-      navItem.href = getRelativeURL(h2a.href);
+  openMenu = (el, isSearch) => {
+    el.classList.add(IS_OPEN);
+    document.addEventListener('click', this.closeOnDocClick);
+    window.addEventListener('keydown', this.closeOnEscape);
+    if (!isSearch) {
+      const desktop = window.matchMedia('(min-width: 1200px)');
+      if (desktop.matches) {
+        document.addEventListener('scroll', this.closeOnScroll, { passive: true });
+      }
+    } else {
+      this.curtain.classList.add(IS_OPEN);
+      el.querySelector('.gnav-search-input').focus();
     }
-    if (div.querySelector('li')) {
-      navItem.submenu = [...div.querySelectorAll('li')].map((li) => {
-        const a = li.querySelector('a');
-        const ni = {
-          text: li.textContent,
-        };
-        if (a) ni.href = getRelativeURL(a.href);
-        return (ni);
-      });
+    this.state.openMenu = el;
+  }
+
+  closeOnScroll = () => {
+    let scrolled;
+    if (!scrolled) {
+      this.toggleMenu(this.state.openMenu);
+      scrolled = true;
+      document.removeEventListener('scroll', this.closeOnScroll);
     }
-    return (navItem);
-  });
-  const logo = nav.top.shift();
-  nav.logo = logo;
-
-  const signInEl = header.querySelector('.sign-in a');
-  if (signInEl) {
-    nav.signIn = {
-      text: signInEl.textContent,
-      href: signInEl.href,
-    };
   }
 
-  const searchEl = header.querySelector('.search a');
-  if (searchEl) {
-    nav.search = {
-      text: searchEl.textContent,
-      href: searchEl.href,
-    };
+  closeOnDocClick = (e) => {
+    const closest = e.target.closest(`.${IS_OPEN}`);
+    const isCurtain = e.target === this.curtain;
+    if ((this.state.openMenu && !closest) || isCurtain) {
+      this.toggleMenu(this.state.openMenu);
+    }
   }
 
-  return nav;
+  closeOnEscape = (e) => {
+    if (e.keyCode === 27) {
```

Comment:
Recommendation is to avoid using this - https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode